### ISSUE TITLE
Create routines for xml production runs

### ISSUE THREAD

**Issue by pochedls** [durack1/CMIPLib#5]
The current dynamic cmip library is still in testing, but it should be refactored and extended to run in several modes (similar to existing cmip xml library production runs). Some ideas include:

- [x] modify code to run from command line with command line options (e.g., report mode)
- [ ] setup sensible cron jobs (pulling from modes below)
- [x] mode to scan paths for new / modified data
- [ ] mode to retry cdscan for xml files with errors (or types of errors)
- [x] functionality to check for deleted paths (against database)
- [ ] functionality to populate denote latest xml 
- [x] functionality to generate statistics on xmls, errors, total number of directories (see #2) 
- [x] mode to scan for a subset of data (e.g., hourly precip data for a given experiment)
- [x] enable functionality to limit scans to directories that are at least a few hours old

Some of these should be broken into separate tickets.

**Comment by durack1** [comment]
@pochedls this is a really well thought out list, nice work!

**Comment by pochedls** [comment]
Just a note to remind ourselves. We should probably limit cdscans to directories that are at least a few hours old (to make sure they aren't actively being updated while the scan is taking place).

**Comment by durack1** [comment]
Agreed, the other idea was the query the database that @painter1 is standing up for the downloads - but this may complicate thing considerably

**Comment by pochedls** [comment]
In c92d9cb406986f96ff232c66c4e57e722e62fc79:
* set up code to run from command line
* parallelized directory search
* option to scan for new/modified paths
* option to create xmls (need to add options)

**Comment by durack1** [comment]
@pochedls I have this in hand, at least the cronjob creation - want me to drop some example code somewhere?
